src/NosCore.GameObject/Group.cs

```diff
@@ -59,31 +59,39 @@ public Group(GroupType type)
 
         public PinitPacket GeneratePinit()
         {
-            var i = 0;
+            var i = 1;
+
+            if (IsEmpty) return new PinitPacket { GroupSize = 0 };
 
             return new PinitPacket
             {
                 GroupSize = Count == 1 ? 0 : Count,
-                PinitSubPackets = Values.Select(s => s.Item2.GenerateSubPinit(Count == 1 ? i : ++i)).ToList()
+                PinitSubPackets = Values.OrderBy(s => s.Item1).Select(s => s.Item2.GenerateSubPinit(Count == 1 ? i : ++i)).ToList()
             };
         }
 
-        public List<PstPacket> GeneratePst()
+        public List<PstPacket> GeneratePst(long visualId)
```

Comment:
no

Reply:
Then how we avoid sending useless info to the same player who "requests" the status of his party? (official avoids sending this info)

Reply:
what do you mean ? the player need to know who is in his party how could it work without ?

Reply:
Sorry for explaining incorrectly. We need to know who is requesting the info because we need to avoid sending his own info. Player A doesn't need to know his own information via "pst" packet, but needs to know Player B and C. This is respected by official.

Reply:
the you need a better name than visualId and also use the VisualEntity instead of the id

Reply:
Okay, will do it when I can!